cluster/router/types/router_plan.go

```diff
@@ -17,49 +17,196 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// RoutingPlanBuildOptions contains parameters used to construct a routing plan
+// for either read or write operations.
+//
+// Fields:
+//   - Shard: The name of the shard to route to. For multi-tenant collections, this must be the tenant name.
+//     For single-tenant collections, this should be empty to route to all shards, or optionally set to a specific shard
+//     if targeting all shards when creating routing plans for reading.
+//   - ConsistencyLevel: The desired level of consistency for the operation.
+//   - DirectCandidateNode: Optional. The preferred node to use first when building the routing plan.
+//     If empty, the local node is used as the default candidate.
 type RoutingPlanBuildOptions struct {
-	Collection             string
-	Shard                  string
-	ConsistencyLevel       ConsistencyLevel
-	DirectCandidateReplica string
+	Shard               string
+	ConsistencyLevel    ConsistencyLevel
+	DirectCandidateNode string
 }
 
-func (r RoutingPlanBuildOptions) Validate() error {
-	if r.Collection == "" {
-		return fmt.Errorf("no collection specified for routing plan building")
-	}
-	if r.Shard == "" {
-		return fmt.Errorf("no shard specified for routing plan building")
-	}
-	return nil
+// String returns a human-readable representation of the RoutingPlanBuildOptions.
+// Useful for debugging and logging.
+func (o RoutingPlanBuildOptions) String() string {
+	return fmt.Sprintf(
+		"RoutingPlanBuildOptions{shard: %q, consistencyLevel: %s, directCandidateNode: %q}",
+		o.Shard, o.ConsistencyLevel, o.DirectCandidateNode,
+	)
 }
 
-type RoutingPlan struct {
-	Collection string
-	Shard      string
-	Replicas   []string
-
+// ReadRoutingPlan represents the plan for routing a read operation.
+//
+// Fields:
+//   - Shard: The (optional) shard targeted by this routing plan. If empty, all relevant shards are targeted.
+//   - ReplicaSet: The ordered list of replicas to contact.
+//   - ConsistencyLevel: The user-specified consistency level.
+//   - IntConsistencyLevel: The resolved numeric value for the consistency level.
+type ReadRoutingPlan struct {
+	Shard               string
```

Comment:
should the collection name be preserved here for traceability in logs or diagnostics, even if it’s not functionally necessary?